metricflow-semantics/metricflow_semantics/specs/linkable_spec_set.py

```diff
@@ -145,14 +139,57 @@ def __len__(self) -> int:  # noqa: D105
         return len(self.dimension_specs) + len(self.time_dimension_specs) + len(self.entity_specs)
 
     @staticmethod
-    def create_from_spec_set(spec_set: InstanceSpecSet) -> LinkableSpecSet:  # noqa: D102
-        return LinkableSpecSet(
-            dimension_specs=spec_set.dimension_specs,
-            time_dimension_specs=spec_set.time_dimension_specs,
-            entity_specs=spec_set.entity_specs,
-            group_by_metric_specs=spec_set.group_by_metric_specs,
-        )
+    def create_from_specs(specs: Sequence[LinkableInstanceSpec]) -> LinkableSpecSet:  # noqa: D102
+        return _group_specs_by_type(specs)
 
-    @staticmethod
-    def create_from_specs(specs: Sequence[InstanceSpec]) -> LinkableSpecSet:  # noqa: D102
-        return LinkableSpecSet.create_from_spec_set(InstanceSpecSet.create_from_specs(specs))
+
+@dataclass
+class _GroupSpecByTypeVisitor(InstanceSpecVisitor[None]):
```

Comment:
Curious what it means to have this underscore prefix on a class?

Reply:
It indicates more of a "private" class that's not mean to be used elsewhere.